service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderProgressionService.java

```diff
@@ -51,13 +65,29 @@ private void progressDraftCaseManagementOrder(CaseDetails caseDetails, CaseManag
         }
     }
 
-    private void progressActionCaseManagementOrder(CaseDetails caseDetails, CaseManagementOrder order) {
-        switch (order.getAction().getType()) {
+    private void progressActionCaseManagementOrder(CaseDetails caseDetails, CaseData caseData, List<String> errors) {
+        switch (caseData.getCmoToAction().getAction().getType()) {
             case SEND_TO_ALL_PARTIES:
-                //TODO: logic for send to all parties, implemented by FPLA-961.
+                LocalDateTime date = hearingBookingService
+                    .getHearingBookingByUUID(caseData.getHearingDetails(), caseData.getCmoToAction().getId())
+                    .getStartDate();
+
+                if (date.isAfter(time.now())) {
```

Comment:
Just for my own learning, what does `time.now` give us that `localDate.now` does not?

Reply:
it's an interface that Joe and Lewis made. It's easier to mock in test essentially.